bin/reth-bench/src/bench/new_payload_only.rs

```diff
@@ -0,0 +1,194 @@
+
+/// `reth benchmark new-payload-only` command
+#[derive(Debug, Parser)]
+pub struct Command {
+    /// The RPC url to use for getting data.
+    #[arg(long, value_name = "RPC_URL", verbatim_doc_comment)]
+    rpc_url: String,
+
+    #[command(flatten)]
+    benchmark: BenchmarkArgs,
+}
+
+impl Command {
+    /// Execute `benchmark new-payload-only` command
+    pub async fn execute(self, _ctx: CliContext) -> eyre::Result<()> {
+        info!("Running benchmark using data from RPC URL: {}", self.rpc_url);
+
+        // set up alloy client for blocks
+        let block_provider = ProviderBuilder::new().on_http(self.rpc_url.parse()?);
+
+        // If neither `--from` nor `--to` are provided, we will run the benchmark continuously,
+        // starting at the latest block.
+        let mut benchmark_mode = match (self.benchmark.from, self.benchmark.to) {
+            (Some(from), Some(to)) => BenchMode::Range(from..=to),
+            (None, None) => BenchMode::Continuous,
+            _ => {
+                // both or neither are allowed, everything else is ambiguous
+                return Err(eyre::eyre!(
+                    "Both --benchmark.from and --benchmark.to must be provided together"
+                ))
+            }
+        };
+
+        // construct the authenticated provider
+        let auth_jwt = self.benchmark.auth_jwtsecret.ok_or_else(|| {
+            eyre::eyre!("--auth-jwtsecret must be provided for authenticated RPC")
+        })?;
+
+        // fetch jwt from file
+        //
+        // the jwt is hex encoded so we will decode it after
+        let jwt = std::fs::read_to_string(auth_jwt)?;
+        let jwt = JwtSecret::from_hex(jwt)?;
+
+        // get engine url
+        let auth_url = Url::parse(&self.benchmark.engine_rpc_url)?;
+
+        // construct the authed transport
+        info!("Connecting to Engine RPC at {} for replay", auth_url);
+        let auth_transport = AuthenticatedTransportConnect::new(auth_url, jwt);
+        let client = ClientBuilder::default().connect_boxed(auth_transport).await?;
+        let auth_provider = RootProvider::<_, AnyNetwork>::new(client);
+
+        let first_block = match benchmark_mode {
+            BenchMode::Continuous => {
+                // fetch Latest block
+                block_provider.get_block_by_number(BlockNumberOrTag::Latest, true).await?.unwrap()
+            }
+            BenchMode::Range(ref mut range) => {
+                match range.next() {
+                    Some(block_number) => {
+                        // fetch first block in range
+                        block_provider
+                            .get_block_by_number(block_number.into(), true)
+                            .await?
+                            .unwrap()
+                    }
+                    None => {
+                        // return an error
+                        panic!("RangeEmpty");
+                    }
+                }
+            }
+        };
```

Comment:
please make a method of `BenchMode`

Reply:
Should be only used once now that the `BenchContext` makes benchmark initialization code shared